### PR TITLE
Rover Skid steering fixs

### PR DESCRIPTION
**Pull request by khancyr**
This address some fix for skid steering configuration : 
- add a parameter to allow more power on on spot turn
- enable pivot turn in auto/ guided
- fix a bug in disarm logic 
- fix skid_steer_in configuration

### CODE REVIEW

**Comment by rmackay9** [comment]
By the way Nick reported the rover drove backwards when pivot steering was enabled in Auto.  I think we need to get to the bottom of that before these changes go into master.

**Comment by NickNunno** [comment]
Yes, this was on Randy's attempt at a quick fix. Rover travels 180* in the opposite direction of the target waypoint. However, it displays proper target/current heading in MP.

**Comment by rmackay9** [comment]
@khancyr, I'm not sure about the skid-steer-in changes.  Can you explain what problem this changes fixes?

I understand the way we do it now may seem round-about in that if a user is using skid-steering input, we turn that into regular steering/throttle values and then we essentially do the reverse within the MotorsUGV library but doing it the way we have it has these advantages:

- keeps the code path simpler because there are fewer exceptions
- keeps input and output very clearly separated
- skid-steering input can be used on regular steering-throttle cars (although I'm not sure why someone would want to do that)

**Comment by khancyr** [comment]
@rmackay9 The skid_steer_in param should allow to move each side independently ! Left joystick will move left side, right joystick will move right side. It is not the case on master currently. And that configuration make only sens to tank style rover (or boat) !

File: APMrover2/AP_MotorsUGV.cpp [hunk 1]
```diff
@@ -268,9 +277,10 @@ void AP_MotorsUGV::output_skid_steering(bool armed, float steering, float thrott
 
     // deal with case of turning on the spot
     if (is_zero(throttle_scaled)) {
-        // full possible range is not used to keep response equivalent to non-zero throttle case
-        motor_left += steering_scaled * 0.5f;
-        motor_right -= steering_scaled * 0.5f;
+        // steering output split evenly between left and right motors and compensated for friction
+        const float friction_comp = MAX(0.0f, 1.0f + (_skid_friction / 100.0f));
```
Comment (khancyr) [inline APMrover2/AP_MotorsUGV.cpp]: well, we could remove the /100.0f for *0.01f ...

**Comment by rmackay9** [comment]
I've merged in all the commits except, "Rover: allow to use skid_steer_in config" so I'm closing this PR.  @khancyr if you'd like to keep pursuing that commit's change can you put it in a new PR?

**Pull request closed without merging**